File: backend/app/schemas/notification.py
from pydantic import BaseModel
from datetime import datetime


class NotificationOut(BaseModel):
    id: str
    user_id: str
    title: str
    message: str | None = None
    notification_type: str
    reference_type: str | None = None
    reference_id: str | None = None
    is_read: bool = False
    created_at: datetime | None = None

    class Config:
        from_attributes = True


class UnreadCountOut(BaseModel):
    count: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: string
  user_id: string
  title: string
  message: string | null
  notification_type: 'workflow' | 'system'
  reference_type: string | null
  reference_id: string | null
  is_read: boolean
  created_at: string | null
}

export interface NotificationListResponse {
  total: number
  page: number
  page_size: number
  items: Notification[]
}

export interface UnreadCountResponse {
  count: number
}